aws/resource_aws_glue_user_defined_function.go

```diff
@@ -0,0 +1,259 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"catalog_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+			},
+			"database_name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				ForceNew:     true,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+			},
+			"class_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+			},
+			"owner_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+			},
+			"owner_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(glue.PrincipalType_Values(), false),
+			},
+			"resource_uris": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1000,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resource_type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(glue.ResourceType_Values(), false),
+						},
+						"uri": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1024),
+						},
+					},
+				},
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueUserDefinedFunctionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+	catalogID := createAwsGlueCatalogID(d, meta.(*AWSClient).accountid)
+	dbName := d.Get("database_name").(string)
+	funcName := d.Get("name").(string)
+
+	input := &glue.CreateUserDefinedFunctionInput{
+		CatalogId:     aws.String(catalogID),
+		DatabaseName:  aws.String(dbName),
+		FunctionInput: expandAwsGlueUserDefinedFunctionInput(d),
+	}
+
+	_, err := conn.CreateUserDefinedFunction(input)
+	if err != nil {
+		return fmt.Errorf("error creating Glue User Defined Function: %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s:%s", catalogID, dbName, funcName))
+
+	return resourceAwsGlueUserDefinedFunctionRead(d, meta)
+}
+
+func resourceAwsGlueUserDefinedFunctionUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).glueconn
+
+	catalogID, dbName, funcName, err := readAwsGlueUDFID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	input := &glue.UpdateUserDefinedFunctionInput{
+		CatalogId:     aws.String(catalogID),
+		DatabaseName:  aws.String(dbName),
+		FunctionName:  aws.String(funcName),
+		FunctionInput: expandAwsGlueUserDefinedFunctionInput(d),
+	}
+
+	if d.HasChanges("owner_name", "owner_type", "class_name", "resource_uris") {
+		if _, err := conn.UpdateUserDefinedFunction(input); err != nil {
+			return err
+		}
+	}
```

Comment:
Can skip `d.HasChanges()` here and should return error messaging 👍 

```suggestion
	if _, err := conn.UpdateUserDefinedFunction(input); err != nil {
		return fmt.Errorf("error updating Glue User Defined Function (%s): %w", d.Id(), err)
	}
```